docs/vignette_parallel_phylo.md

```diff
@@ -0,0 +1,208 @@
+```
+
+
+(sec_vignette_parallel)=
+
+
+# Vignette: Parallelizing SLiM simulations in a phylogenetic tree
+
+Imagine you want to simulate the evolutionary history of a group. If there is
+no migration between any of the branches in your tree, any branches stemming
+from the same node can be simulated in parallel (see {numref}`phylo`).
+
+```{figure} _static/phylo.pdf
+:height: 200px
+:name: phylo
+
+Example of phylogeny we might want to simulate. Note how branches with the same color can be simulated in parallel when there is no migration.
+```
+
+## Simulating the branches
+First, we need to write a SLiM script that will be used for simulating the
+history of each branch in our phylogeny.
+We will perform a simple simulation, in which each branch can have a different
+(but fixed) population size and length (number of generations).
+Also, we will allow deleterious mutations to happen across the entire chromosome
+at a fixed rate.
+
+Here is a SLiM script that would do this:
+
+```{literalinclude} phylo_bgs.slim
+```
+
+For each branch, we will tell SLiM whether we want to start it from a previous
+branch or not, read the previous tree sequence (if necessary) and change the
+population size accordingly.
+Note that when you read a tree sequence into SLiM, the generation counter will
+be updated with the time encoded in the tree sequence, so we need to set the end
+of the simulation as the length of the branch (`num_gens`) plus the total length of the
+loaded tree sequence.
+At the end of the simulation, we call `sim.treeSeqRememberIndividuals` right
+before saving the resulting tree sequence. This is necessary because we need to
+ensure the individuals in the final generation are never dropped from the tree
+sequence in future runs of SLiM which are started from the output of the
+simulation, as they will later be used to glue the tree sequences together.
+
+I encoded the phylogeny we will simulate in a simple table:
+
+```{code-cell}
+:tags: ["hide-input"]
+df = pd.read_csv("_static/phylo.tsv", sep="\t")
+df = df.fillna('')
+df["infile"] = df.parent + ".trees"
+df["outfile"] = df.child + ".trees"
+df.loc[df["infile"]==".trees", "infile"] = ""
+df["is_leaf"] = ~df.child.isin(df.parent)
+df
+```
+
+With our phylogeny and the simulation parameters, we are ready to run our
+simulations.
+One way to parallelize the simulation of sister branches is to use `make`.
+You do not need to know much about this tool (though it is totally worthwhile
+to check it out).
+The main idea here is that you can specify dependency between files and `make`
+works its magic to run the simulations in the right order.
+
+```{code-cell}
+f = open("sims.make", "w")
+print(f"all: {' '.join(df.outfile.to_list())}\n", file=f)
+for i, row in df.iterrows():
+    print(f"{row.outfile}: {row.infile}", file=f)
+    print(f"\tslim -d \"infile='{row.infile}'\" -d popsize={row.popsize} -d num_gens={row.edgelen} -d \"outfile='{row.child}.trees'\" phylo_bgs.slim\n", file=f)
```

Comment:
I love it! We could instead do 
```
     print(f"\tslim -d \"infile='$<\" -d popsize={row.popsize} -d num_gens={row.edgelen} -d \"outfile='$@'\" phylo_bgs.slim\n", file=f)
```
... but I suppose that'd be less transparent.

Reply:
ah, yeah, totally forgot about those make wildcards. though I agree it would make this a bit harder to read.